### PR TITLE
tests diagnostics timestamps arange

### PR DESCRIPTION
**Pull request by PhilipDeegan**
functional test validation issues

### CODE REVIEW

**Comment by PhilipDeegan** [comment]
conservation test is still running but is not edited in this PR (so far)
image

File: tests/functional/translation/translat1d.py [hunk 1]
```diff
@@ -87,7 +88,7 @@ def vthz(x):
 
     sim = ph.global_vars.sim
 
-    timestamps = np.arange(0, sim.final_time +sim.time_step, sim.time_step)
+    timestamps = sim.time_step * np.arange(0, sim.final_time / sim.time_step + 1)
```
Comment (PhilipDeegan) [inline tests/functional/translation/translat1d.py:91]: missed one

File: tests/diagnostic/__init__.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
 
 
+def all_timestamps(sim):
+    import numpy as np
+    return sim.time_step * np.arange(0, sim.final_time / sim.time_step + 1)
+
```
Comment (PhilipDeegan) [inline tests/diagnostic/__init__.py:6]: maybe better
```
def timestamps_with_step(sim, dump_step):
    import numpy as np
    nbr_dump_step = sim.final_time / dump_step + 1
    dump_step*np.arange(0, nbr_dump_step)

def all_timestamps(sim):
    return timestamps_with_step(sim, sim.time_step)
```

**Review by nicolasaunai — APPROVED** [review]
what's up wiht ubsan?

**Comment by PhilipDeegan** [comment]
> what's up wiht ubsan?

duplicate entry 

image

**Pull request merged**